docs/source/concepts.rst

```diff
@@ -254,6 +260,7 @@ These events could be used to attach any handler and are fired using :meth:`~ign
     @engine.on(Events.STARTED)
     def fire_custom_events(engine):
          engine.fire_event(CustomEvents.CUSTOM_STARTED)
+         engine.state.custom_started += 1  # increase state counter for this event (required)
```

Comment:
Let's say this is required for filtering only and can be omitted if no filtering is done. And another note to add is that this is a temporary solution and subject to change in future releases.